flixel/system/render/hardware/gl/HardwareRenderer.hx

```diff
@@ -0,0 +1,322 @@
+		
+		if (needRenderHelper)
+		{
+			renderHelper.capture(false);
+			uMatrix = renderHelper.getMatrix(transform, renderer, numPasses);
+		}
+		else
+		{
+			var matrix = renderer.getMatrix(transform);
+			uMatrix = GLRenderHelper.matrixToArray(matrix);
+		}
+		
+		var worldColor:ColorTransform = this.__worldColorTransform;
+		
+		uColor[0] = worldColor.redMultiplier;
+		uColor[1] = worldColor.greenMultiplier;
+		uColor[2] = worldColor.blueMultiplier;
+		uColor[3] = this.__worldAlpha;
+		
+		var shader:FlxShader = null;
+		var nextShader:FlxShader = null;
+		var blend:BlendMode = null;
+		var texture:FlxGraphic = null;
+		
+		var i:Int = 0;
+		
+		while (i < stateNum)
+		{
+			var state:FlxDrawHardwareItem<Dynamic> = states[i];
+			
+			nextShader = (state.graphics != null) ? texturedTileShader : coloredTileShader;
+			nextShader = (state.shader != null) ? state.shader : nextShader;
+			
+			if (shader != nextShader || shader == null)
+			{
+				shader = nextShader;
+				
+				shader.data.uMatrix.value = uMatrix;
+				shader.data.uColor.value = uColor;
+				
+				renderSession.shaderManager.setShader(shader);
+			}
+			
+			gl.uniform4f(shader.data.uColorOffset.index, state.redOffset, state.greenOffset, state.blueOffset, state.alphaOffset);
+			
+			if (blend != state.blending)
+			{
+				renderSession.blendModeManager.setBlendMode(state.blending);
+				blend = state.blending;
+			}
+			
+			if (texture != state.graphics)
+			{
+				texture = state.graphics;
+				
+				if (texture != null)
+				{
+					gl.bindTexture(gl.TEXTURE_2D, texture.bitmap.getTexture(gl));
+				}
+			}
+			
+			if (state.glBuffer == null)
+			{
+				state.glBuffer = gl.createBuffer();
+				#if !FLX_RENDER_GL_ARRAYS
+				state.glIndexes = gl.createBuffer();
+				#end
+			}
+			
+			#if !FLX_RENDER_GL_ARRAYS
+			gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, state.glIndexes);
+			#end
+			
+			#if !FLX_RENDER_GL_ARRAYS
+			if (state.indexBufferDirty)
+			{
+				state.indexBufferDirty = false;
+				gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, state.indexes, gl.DYNAMIC_DRAW);
+			}
+			#end
+			
+			gl.bindBuffer(gl.ARRAY_BUFFER, state.glBuffer);
+			
+			if (state.vertexBufferDirty)
+			{
+				state.vertexBufferDirty = false;
+				gl.bufferData(gl.ARRAY_BUFFER, state.buffer, gl.DYNAMIC_DRAW);
+			}
+			
+			var stride:Int = state.elementsPerVertex * Float32Array.BYTES_PER_ELEMENT;
+			var offset:Int = 0;
+			
+			gl.vertexAttribPointer(shader.data.aPosition.index, 2, gl.FLOAT, false, stride, offset * Float32Array.BYTES_PER_ELEMENT);
+			offset += 2;
+			
+			if (texture != null)
+			{
+				// texture smoothing
+				if (state.antialiasing) 
+				{
+					gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+					gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);	
+				} 
+				else 
+				{
+					gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+					gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+				}
+				
+				#if !js
```

Comment:
yes, it causes errors on js target. something like incorrect parameter value.
i just want to be sure that texture repeating is on.